src/commonTest/kotlin/com/charleskorn/kaml/YamlReadingTest.kt

```diff
@@ -2079,7 +2079,7 @@ class YamlReadingTest : FlatFunSpec({
                     override val descriptor: SerialDescriptor
                         get() = String.serializer().descriptor
 
-                    override fun deserialize(decoder: Decoder): Inner = Inner("from context serializer")
+                    override fun deserialize(decoder: Decoder): Inner = Inner(decoder.decodeString())
```

Comment:
How about something like this, so that we can be confident the contextual serializer was used?

```suggestion
                    override fun deserialize(decoder: Decoder): Inner = Inner("from context serializer: " + decoder.decodeString())
```

Reply:
There is no other serializer that can be used because `Inner` class is not annotated with `@Serializable`. But I don't see any harm if we modify the test as you suggested)